src/server/api/routers/book.ts

```diff
@@ -0,0 +1,27 @@
+import { z } from "zod";
+
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+
+export const bookRouter = createTRPCRouter({
+  create: protectedProcedure
+    .input(z.object({ title: z.string().min(1), author: z.string().min(1) }))
+    .mutation(async ({ ctx, input }) => {
+      // simulate a slow db call
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      return ctx.db.book.create({
+        data: {
+          title: input.title,
+          author: input.author,
+          createdBy: { connect: { id: ctx.session.user.id } },
```

Comment:
alternative here is to write `createdById: ctx.serssion.user.id`, which is a bit tighter